src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryPath.java

```diff
@@ -33,7 +33,11 @@ public class BigQueryPath {
   private String dataset;
   private String table;
   private static final int NAME_MAX_LENGTH = 1024;
-  private static final String VALID_NAME_REGEX = "[\\w]+";
+  // Valid BigQuery dataset names can contain only letters, numbers, and underscores.
+  // See here: https://cloud.google.com/bigquery/docs/datasets#dataset-naming
+  private static final String VALID_DATASET_NAME_REGEX = "[\\w]+";
```

Comment:
All regex should be of type `Pattern` so that we have it compiled and reused.

```
Pattern VALID_DATESET_NAME = Pattern.compile("\\w+");
```

Reply:
Fixed! I also made the equivalent changes in SpannerPath.java, and in my bigquery-delta-plugins PR